gpkit/constraints/sgp.py

```diff
@@ -108,6 +105,8 @@ def localsolve(self, solver=None, verbosity=1, x0=None, reltol=1e-4,
             print("Beginning signomial solve.")
         self.gps = []  # NOTE: SIDE EFFECTS
         self.results = []
+        if x0 and mutategp:
+            self._gp = self.init_gp(self.substitutions, verbosity, x0)
```

Comment:
It doesn't because in `.gp()` the x0 is not fully updated if `mutategp` is True; instead it relies on `init_gp` having been called with the full x0 first